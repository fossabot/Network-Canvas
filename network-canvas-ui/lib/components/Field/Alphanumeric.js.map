{"version":3,"sources":["../../../src/components/Field/Alphanumeric.js"],"names":["Alphanumeric","field","label","input","meta","invalid","error"],"mappings":";;;;;;;;AAAA;;;;;;AAEA;;;AAGA,IAAMA,eAAe,SAAfA,YAAe;AAAA,SACnB;AAAA;AAAA;AACE,sDAAO,MAAK,MAAZ,EAAmB,aAAaC,MAAMC,KAAtC,IAAiDD,MAAME,KAAvD,EADF;AAEGF,UAAMG,IAAN,CAAWC,OAAX,IACC;AAAA;AAAA;AAAMJ,YAAMG,IAAN,CAAWE;AAAjB;AAHJ,GADmB;AAAA,CAArB;;kBAQeN,Y","file":"Alphanumeric.js","sourcesContent":["import React from 'react';\n\n/**\n  * Alphanumeric field type\n  */\nconst Alphanumeric = field => (\n  <div>\n    <input type=\"text\" placeholder={field.label} {...field.input} />\n    {field.meta.invalid &&\n      <div>{field.meta.error}</div>}\n  </div>\n);\n\nexport default Alphanumeric;\n"]}