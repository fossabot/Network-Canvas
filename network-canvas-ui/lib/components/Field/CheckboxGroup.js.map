{"version":3,"sources":["../../../src/components/Field/CheckboxGroup.js"],"names":["CheckboxGroup","onClickOption","clickedOption","props","input","value","onChange","options","option","label","meta","name","invalid","error","propTypes","object","isRequired","array","string"],"mappings":";;;;;;;;;;AAAA;;;;AACA;;;;AACA;;AACA;;;;;;;;;;;;;;AAEA;;;IAGMA,a;;;;;;;;;;;;;;oMACJC,a,GAAgB,UAACC,aAAD,EAAmB;AAAA,8BACM,MAAKC,KADX,CACzBC,KADyB;AAAA,UAChBC,KADgB,qBAChBA,KADgB;AAAA,UACTC,QADS,qBACTA,QADS;;;AAGjCA,4BACK,uBAAU,iBAAI,MAAKH,KAAL,CAAWI,OAAf,EAAwB;AAAA,eAAU,CAACC,MAAD,EAAS,KAAT,CAAV;AAAA,OAAxB,CAAV,CADL,EAEMH,SAAS,EAFf,sBAGQH,aAHR,EAGwB,CAACG,MAAMH,aAAN,CAHzB;AAKD,K;;;;;6BAEQ;AAAA,mBACyC,KAAKC,KAD9C;AAAA,UACCM,KADD,UACCA,KADD;AAAA,UACQC,IADR,UACQA,IADR;AAAA,gCACcN,KADd;AAAA,UACuBO,IADvB,gBACuBA,IADvB;AAAA,UAC6BN,KAD7B,gBAC6BA,KAD7B;;;AAGP,aACE;AAAA;AAAA;AACE;AAAA;AAAA;AAAMI;AAAN,SADF;AAEE;AACE,gBAAME,IADR;AAEE,mBAASN,SAAS,uBAAU,iBAAI,KAAKF,KAAL,CAAWI,OAAf,EAAwB;AAAA,mBAAU,CAACC,MAAD,EAAS,KAAT,CAAV;AAAA,WAAxB,CAAV,CAFpB;AAGE,yBAAe,KAAKP;AAHtB,UAFF;AAOGS,aAAKE,OAAL,IACC;AAAA;AAAA;AAAMF,eAAKG;AAAX;AARJ,OADF;AAYD;;;;;;AAGHb,cAAcc,SAAd,GAA0B;AACxBV,SAAO,oBAAUW,MAAV,CAAiBC,UADA;AAExBT,WAAS,oBAAUU,KAAV,CAAgBD,UAFD;AAGxBP,SAAO,oBAAUS,MAAV,CAAiBF,UAHA;AAIxBN,QAAM,oBAAUK,MAAV,CAAiBC;AAJC,CAA1B;;kBAOehB,a","file":"CheckboxGroup.js","sourcesContent":["import React, { Component } from 'react';\nimport PropTypes from 'prop-types';\nimport { fromPairs, map } from 'lodash';\nimport Checkboxes from './Checkboxes';\n\n/**\n  * A checkbox list that sets thes field value to a key/value object of boolean properties\n  */\nclass CheckboxGroup extends Component {\n  onClickOption = (clickedOption) => {\n    const { input: { value, onChange } } = this.props;\n\n    onChange({\n      ...fromPairs(map(this.props.options, option => [option, false])),\n      ...(value || {}),\n      ...{ [clickedOption]: !value[clickedOption] },\n    });\n  }\n\n  render() {\n    const { label, meta, input: { name, value } } = this.props;\n\n    return (\n      <div>\n        <div>{label}</div>\n        <Checkboxes\n          name={name}\n          options={value || fromPairs(map(this.props.options, option => [option, false]))}\n          onClickOption={this.onClickOption}\n        />\n        {meta.invalid &&\n          <div>{meta.error}</div>}\n      </div>\n    );\n  }\n}\n\nCheckboxGroup.propTypes = {\n  input: PropTypes.object.isRequired,\n  options: PropTypes.array.isRequired,\n  label: PropTypes.string.isRequired,\n  meta: PropTypes.object.isRequired,\n};\n\nexport default CheckboxGroup;\n"]}