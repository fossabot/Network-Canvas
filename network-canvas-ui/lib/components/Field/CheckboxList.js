'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _lodash = require('lodash');

var _Checkboxes = require('./Checkboxes');

var _Checkboxes2 = _interopRequireDefault(_Checkboxes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
  * A checkbox list that sets thes field value to a list of selected options
  */
var CheckboxList = function (_Component) {
  _inherits(CheckboxList, _Component);

  function CheckboxList() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, CheckboxList);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = CheckboxList.__proto__ || Object.getPrototypeOf(CheckboxList)).call.apply(_ref, [this].concat(args))), _this), _this.onClickOption = function (option) {
      var _this$props$input = _this.props.input,
          value = _this$props$input.value,
          onChange = _this$props$input.onChange;


      if (value.indexOf(option) === -1) {
        onChange((0, _lodash.sortBy)([].concat(_toConsumableArray(value), [option])));
      } else {
        onChange((0, _lodash.without)(value, option));
      }
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(CheckboxList, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          label = _props.label,
          options = _props.options,
          meta = _props.meta,
          _props$input = _props.input,
          name = _props$input.name,
          value = _props$input.value;


      var checks = (0, _lodash.reduce)(options, function (memo, option) {
        return _extends({}, memo, _defineProperty({}, option, value.indexOf(option) !== -1));
      }, {});

      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          'div',
          null,
          label
        ),
        _react2.default.createElement(_Checkboxes2.default, { name: name, options: checks, onClickOption: this.onClickOption }),
        meta.invalid && _react2.default.createElement(
          'div',
          null,
          meta.error
        )
      );
    }
  }]);

  return CheckboxList;
}(_react.Component);

CheckboxList.propTypes = {
  input: _propTypes2.default.object.isRequired,
  options: _propTypes2.default.array.isRequired,
  label: _propTypes2.default.string.isRequired,
  meta: _propTypes2.default.object.isRequired
};

exports.default = CheckboxList;
module.exports = exports['default'];
//# sourceMappingURL=CheckboxList.js.map