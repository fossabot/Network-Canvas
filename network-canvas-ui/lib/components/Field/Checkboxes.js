'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _lodash = require('lodash');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
  * A list of checkboxes, not intended for direct use
  */
var Checkboxes = function Checkboxes(_ref) {
  var name = _ref.name,
      options = _ref.options,
      onClickOption = _ref.onClickOption;
  return _react2.default.createElement(
    'div',
    null,
    (0, _lodash.map)(options, function (value, option) {
      return _react2.default.createElement(
        'div',
        { key: option },
        _react2.default.createElement(
          'label',
          { htmlFor: name + '_' + option },
          _react2.default.createElement('input', {
            type: 'checkbox',
            id: name + '_' + option,
            name: name,
            value: option,
            checked: value,
            onClick: function onClick() {
              onClickOption(option);
            }
          }),
          ' ',
          option
        )
      );
    })
  );
};

Checkboxes.propTypes = {
  name: _propTypes2.default.string.isRequired,
  options: _propTypes2.default.array.isRequired,
  onClickOption: _propTypes2.default.func.isRequired
};

exports.default = Checkboxes;
module.exports = exports['default'];
//# sourceMappingURL=Checkboxes.js.map