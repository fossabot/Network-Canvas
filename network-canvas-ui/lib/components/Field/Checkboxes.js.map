{"version":3,"sources":["../../../src/components/Field/Checkboxes.js"],"names":["Checkboxes","name","options","onClickOption","value","option","propTypes","string","isRequired","array","func"],"mappings":";;;;;;AAAA;;;;AACA;;;;AACA;;;;AAEA;;;AAGA,IAAMA,aAAa,SAAbA,UAAa;AAAA,MAAGC,IAAH,QAAGA,IAAH;AAAA,MAASC,OAAT,QAASA,OAAT;AAAA,MAAkBC,aAAlB,QAAkBA,aAAlB;AAAA,SACjB;AAAA;AAAA;AACG,qBAAID,OAAJ,EAAa,UAACE,KAAD,EAAQC,MAAR;AAAA,aACZ;AAAA;AAAA,UAAK,KAAKA,MAAV;AACE;AAAA;AAAA,YAAO,SAAYJ,IAAZ,SAAoBI,MAA3B;AACE;AACE,kBAAK,UADP;AAEE,gBAAOJ,IAAP,SAAeI,MAFjB;AAGE,kBAAMJ,IAHR;AAIE,mBAAOI,MAJT;AAKE,qBAASD,KALX;AAME,qBAAS,mBAAM;AAAED,4BAAcE,MAAd;AAAwB;AAN3C,YADF;AAAA;AAQMA;AARN;AADF,OADY;AAAA,KAAb;AADH,GADiB;AAAA,CAAnB;;AAmBAL,WAAWM,SAAX,GAAuB;AACrBL,QAAM,oBAAUM,MAAV,CAAiBC,UADF;AAErBN,WAAS,oBAAUO,KAAV,CAAgBD,UAFJ;AAGrBL,iBAAe,oBAAUO,IAAV,CAAeF;AAHT,CAAvB;;kBAMeR,U","file":"Checkboxes.js","sourcesContent":["import React from 'react';\nimport PropTypes from 'prop-types';\nimport { map } from 'lodash';\n\n/**\n  * A list of checkboxes, not intended for direct use\n  */\nconst Checkboxes = ({ name, options, onClickOption }) => (\n  <div>\n    {map(options, (value, option) => (\n      <div key={option}>\n        <label htmlFor={`${name}_${option}`}>\n          <input\n            type=\"checkbox\"\n            id={`${name}_${option}`}\n            name={name}\n            value={option}\n            checked={value}\n            onClick={() => { onClickOption(option); }}\n          /> {option}\n        </label>\n      </div>\n    ))}\n  </div>\n);\n\nCheckboxes.propTypes = {\n  name: PropTypes.string.isRequired,\n  options: PropTypes.array.isRequired,\n  onClickOption: PropTypes.func.isRequired,\n};\n\nexport default Checkboxes;\n"]}