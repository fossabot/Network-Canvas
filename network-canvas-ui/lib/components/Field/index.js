'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Numeric = require('./Numeric');

Object.defineProperty(exports, 'Numeric', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Numeric).default;
  }
});

var _Alphanumeric = require('./Alphanumeric');

Object.defineProperty(exports, 'Alphanumeric', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Alphanumeric).default;
  }
});

var _CheckboxList = require('./CheckboxList');

Object.defineProperty(exports, 'CheckboxList', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_CheckboxList).default;
  }
});

var _CheckboxGroup = require('./CheckboxGroup');

Object.defineProperty(exports, 'CheckboxGroup', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_CheckboxGroup).default;
  }
});

var _ToggleGroup = require('./ToggleGroup');

Object.defineProperty(exports, 'ToggleGroup', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_ToggleGroup).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
//# sourceMappingURL=index.js.map