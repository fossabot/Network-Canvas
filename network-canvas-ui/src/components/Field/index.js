export { default as Numeric } from './Numeric';
export { default as Alphanumeric } from './Alphanumeric';
export { default as CheckboxList } from './CheckboxList';
export { default as CheckboxGroup } from './CheckboxGroup';
export { default as ToggleGroup } from './ToggleGroup';
