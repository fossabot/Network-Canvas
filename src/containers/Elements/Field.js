/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';
import { Field as ReduxFormField } from 'redux-form';
import { map, toPairs } from 'lodash';
import validations from '../../utils/Validations';
import components from '../../components/Inputs/fieldComponents';
import { withOptionsFromSelector } from '../../behaviours';

/**
  * Returns the named field compontent, if no matching one is found it returns an empty one.
  * @param {string} type The name of the field component to return.
  */
const getComponent = type => (
  Object.hasOwnProperty.call(components, type) ?
  components[type] :
  () => (<div>Field type not defined</div>)
);

/**
* Returns the named validation function, if no matching one is found it returns a validation
* which will always fail.
* @param {string} validation The name of the validation function to return.
  */
const getValidation = validation =>
  map(
    toPairs(validation),
    ([type, options]) => (
      Object.hasOwnProperty.call(validations, type) ? validations[type](options) : () => (`Validation "${type}" not found`)
    ),
  );

/**
  * Renders a redux-form field in the style of our app.
  * @param {string} label Presentational label
  * @param {string} name Property name
  * @param {string} type Field component type
  * @param {string} placeholder Presentational placeholder text
  * @param {object} validation Validation methods
  */
const Field = ({ label, name, type, validation, optionsSelector, ...rest }) => {
  const validate = getValidation(validation);
  let component = getComponent(type);
  if (optionsSelector) { component = withOptionsFromSelector(component, optionsSelector); }
  return (
    <ReduxFormField
      name={name}
      label={label}
      component={component}
      validate={validate}
      {...rest}
    />
  );
};

Field.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  optionsSelector: PropTypes.func,
  validation: PropTypes.object,
};

Field.defaultProps = {
  validation: {},
  optionsSelector: null,
};

export default Field;
