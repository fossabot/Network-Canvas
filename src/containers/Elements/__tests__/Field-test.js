/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import Field from '../Field';
import FieldComponents from '../../../components/Inputs/fieldComponents';

const attributes = {
  label: 'Name',
  name: 'name',
  type: 'Alphanumeric',
};

const validation = {
  required: true,
  minLength: 2,
};

describe('Containers/Elements/Field', () => {
  it('Loads component from the register', () => {
    const field = shallow(<Field {...attributes} />);

    expect(field.find('Field').prop('component')).toBe(FieldComponents.Alphanumeric);
  });

  it('Loads validations from the register', () => {
    const field = shallow(<Field {...attributes} validation={validation} />);

    const validate = field.find('Field').prop('validate');

    expect(validate.length).toBe(2);
  });

  it('Loads options with optionsSelector');
});
