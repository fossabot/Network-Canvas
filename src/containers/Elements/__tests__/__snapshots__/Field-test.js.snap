// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers/Elements/Field should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Field
    component={[Function]}
    label="Name"
    name="name"
    optionsSelector={null}
    type="Alphanumeric"
    validate={Array []}
/>,
  "nodes": Array [
    <Field
      component={[Function]}
      label="Name"
      name="name"
      optionsSelector={null}
      type="Alphanumeric"
      validate={Array []}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Field
        label="Name"
        name="name"
        optionsSelector={null}
        type="Alphanumeric"
        validation={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "label": "Name",
          "name": "name",
          "optionsSelector": null,
          "type": "Alphanumeric",
          "validation": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Field
          component={[Function]}
          label="Name"
          name="name"
          optionsSelector={null}
          type="Alphanumeric"
          validate={Array []}
/>,
        "_debugID": 2,
        "_renderedOutput": <Field
          component={[Function]}
          label="Name"
          name="name"
          optionsSelector={null}
          type="Alphanumeric"
          validate={Array []}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Field
    label="Name"
    name="name"
    optionsSelector={null}
    type="Alphanumeric"
    validation={Object {}}
/>,
}
`;
